lib/tasks/queue.rake

```diff
@@ -29,31 +30,35 @@ namespace :queue do
   end
 
   desc "Add published editions to the message queue by document type"
-  task :requeue_document_type, [:document_type] => :environment do |_, args|
-    document_type = args[:document_type]
-    raise ValueError("expecting document_type") unless document_type.present?
+  task :requeue_document_type, [] => :environment do |_, args|
+    document_types = args.extras
 
-    # Restrict scope to stuff that's live (published or unpublished)
-    # Unpublished content without a content store representation won't
-    # be returned, but we're not interested in this content.
-    scope = Edition
-      .with_document
-      .with_unpublishing
-      .where(document_type: document_type)
-      .where(content_store: :live)
-      .select(:id)
+    raise(ValueError, "expecting document_type") unless document_types.present?
 
-    version = Event.maximum(:id)
+    document_types.each do |document_type|
+
+      # Restrict scope to stuff that's live (published or unpublished)
+      # Unpublished content without a content store representation won't
+      # be returned, but we're not interested in this content.
+      scope = Edition
+        .with_document
+        .with_unpublishing
```

Comment:
I know that you're just moving this code, but as far as I can see, the join on the unpublishings table isn't used.

Reply:
Yep I think you're right. It was originally there as a way of filtering published+withdrawn content, but the `content_store: live` filter handles this now.

Reply:
@cbaines why wouldn't it be used?

Reply:
It's an outer join but I don't think is anything using values from `unpublishings` (correct me if I'm wrong)

Reply:
Indeed, I was also assuming that `with_unpublishing` was doing an outer join on the unpublishings table, and if you're just selecting the edition id, then doing (or not doing) that outer join shouldn't make a difference to the intentional behaviour of the code.

Reply:
👍